execute/exectypes/observation.go

```diff
@@ -126,6 +127,22 @@ func (td TokenData) IsReady() bool {
 	return td.Ready
 }
 
+// MessageTokenID is a unique identifier for a message token data (per chain selector). It's a composite key of
+// the message sequence number and the token index within the message. It's used to easier identify token data for
+// messages without having to deal with nested maps.
+type MessageTokenID struct {
+	SeqNr cciptypes.SeqNum
+	Index int
+}
+
+func NewMessageTokenID(seqNr cciptypes.SeqNum, index int) MessageTokenID {
+	return MessageTokenID{SeqNr: seqNr, Index: index}
+}
+
+func (mti MessageTokenID) String() string {
+	return fmt.Sprintf("%d_%d", mti.SeqNr, mti.Index)
+}
+
```

Comment:
There are over 100 lines of token data types, maybe worth putting them in their own file.

At this point, it may also be helpful to have some sort of builder to manage working with this complex object. That should allow you to stick with the unwieldy triple-map type without hurting the business logic code so much.

Reply:
Actually, I wanted to move away from this triple map monster. I guess it's easier to use SeqNr/Index identifier as a single item identifying the token, instead of using these maps everywhere. That saves one nested for loop during the processing ;)

Reply:
Anything to simplify the usage of this thing is great, just want to make sure we don't lose type safety. may still be worth moving these to a separate file.

Reply:
Agree, already moved